cmd/util/ledger/migrations/storage_v6.go

```diff
@@ -1,3744 +1,3744 @@
-	fields := make([]newInter.CompositeField, 0)
-
-	value.Fields().Foreach(func(key string, fieldVal oldInter.Value) {
-		newValue := c.Convert(fieldVal, nil)
-		if newValue != nil {
-			fields = append(
-				fields,
-				newInter.CompositeField{
-					Name:  key,
-					Value: newValue,
-				},
-			)
-		}
-	})
-
-	c.result = newInter.NewCompositeValue(
-		c.newInter,
-		compositeTypeLocation(value.Location()),
-		value.QualifiedIdentifier(),
-		value.Kind(),
-		fields,
-		*value.Owner,
-	)
-
-	// Do not descent
-	return false
-}
-
-func compositeTypeLocation(location common.Location) common.Location {
-	addressLocation, ok := location.(common.AddressLocation)
-	if !ok {
-		return location
-	}
-
-	addressHex := addressLocation.Address.Hex()
-	switch {
-	case strings.HasPrefix(addressLocation.Name, "FlowIDTableStaking"):
-		switch addressHex {
-		case "e94f751ba094ef6a",
-			"ecda6c5746d5bdf0",
-			"f1a43bfd1354c9b8",
-			"16a5fe3b527633d4",
-			"76d9ea44cef09e20",
-			"9798362e92e5539a":
-			address, err := hex.DecodeString("9eca2b38b18b5dfe")
-			if err != nil {
-				panic(err)
-			}
-
-			location = common.AddressLocation{
-				Address: common.BytesToAddress(address),
-				Name:    addressLocation.Name,
-			}
-		}
-	case strings.HasPrefix(addressLocation.Name, "KittyItems"):
-		switch addressHex {
-		case "fcceff21d9532b58",
-			"17341c7824b030be",
-			"f79ee844bfa76528":
-			address, err := hex.DecodeString("8c5244250369a9ce")
-			if err != nil {
-				panic(err)
-			}
-
-			location = common.AddressLocation{
-				Address: common.BytesToAddress(address),
-				Name:    addressLocation.Name,
-			}
-		}
-	case addressLocation.Name == "DisruptNowBeta":
-		if addressHex == "45888dabccc5c376" {
-			address, err := hex.DecodeString("849832a65c0524b5")
-			if err != nil {
-				panic(err)
-			}
-
-			location = common.AddressLocation{
-				Address: common.BytesToAddress(address),
-				Name:    addressLocation.Name,
-			}
-		}
-	case addressLocation.Name == "NonFungibleToken":
-		if addressHex == "cd2fde7d198629e4" {
-			address, err := hex.DecodeString("c2f5b3fb0ad43ff1")
-			if err != nil {
-				panic(err)
-			}
-
-			location = common.AddressLocation{
-				Address: common.BytesToAddress(address),
-				Name:    addressLocation.Name,
-			}
-		}
-	case addressLocation.Name == "NonFungibleBeatoken":
-		if addressHex == "70239ed8e4c7367a" {
-			address, err := hex.DecodeString("14f7f8198e156fb0")
-			if err != nil {
-				panic(err)
-			}
-
-			location = common.AddressLocation{
-				Address: common.BytesToAddress(address),
-				Name:    addressLocation.Name,
-			}
-		}
-
-	// mainnet
-	case addressLocation.Name == "LockedTokens":
-		if addressHex == "31aed847945124fd" {
-			address, err := hex.DecodeString("8d0e87b65159ae63")
-			if err != nil {
-				panic(err)
-			}
-
-			location = common.AddressLocation{
-				Address: common.BytesToAddress(address),
-				Name:    addressLocation.Name,
-			}
```

Comment:
@SupunS I assume if we don't fix these types we are still fine right if we use older cadence version?

Reply:
Yes, that should be fine. These are still broken for a program even now (I think they are never accessed). We only needed to fix those so that the conversion from old to new storage format works and no values are dropped in the process.